packages/next/src/client/components/react-dev-overlay/ui/components/overview/segment-explorer.tsx

```diff
@@ -71,19 +33,24 @@ function PageSegmentTree({ tree }: { tree: Trie<SegmentNode> | undefined }) {
 function PageSegmentTreeLayerPresentation({
   tree,
   segment,
+  parentSegment,
   node,
   level,
+  nodeCountRef,
 }: {
   tree: Trie<SegmentNode>
   segment: string
+  parentSegment: string
   node: TrieNode<SegmentNode>
   level: number
+  nodeCountRef: React.RefObject<number>
```

Comment:
How does the component know when this value changes? You're also incrementing this value during render and side-effects in render break React's rendering model